### PR TITLE
fix(docs): fix outdated documentation and simplify explanations

### PR DESCRIPTION
**Pull request by adaki2004**
Fix outdated documentation and simplify explanations.
As per: https://github.com/taikoxyz/website/issues/1

### CODE REVIEW

File: packages/website/pages/docs/concepts/overview.mdx [hunk 1]
```diff
@@ -71,40 +71,19 @@ to pay for a proof is not obvious however:
...
+
+Upon a proposer's submission of a block, the signature granted by the chosen provider is subjected to verification. Any deviations result in a reverted transaction.
+
+As an additional incentive for proposers, the system incorporates the issuance of TKO tokens. This serves as an extra motivator, as proposing blocks alone may not always prove profitable, especially when considering Ethereum's on-chain fees plus the proving fee. The issuance of TKO tokens operates on a dynamic 'emission rate per second,' comparing each block proposal to the last.
+
+The reward depends on the proof service provider (and the agreement). For EOAs and Prover pools that implement the IERC1271 interface, the reward is disbursed in ETH. However, in cases where providers implement the IProver interface, the prover fee can be ETH, any other ERC20 tokens, or even NFTs, based on the negotiated terms.
```
Comment (dantaik) [inline packages/website/pages/docs/concepts/overview.mdx:84]:
```suggestion
The reward depends on the proof service provider and the agreement. For EOAs and Prover pools that implement the IERC1271 interface, the reward is disbursed in ETH. However, in cases where providers implement the IProver interface, the prover fee can be ETH, any other ERC20 tokens, or even NFTs, based on the negotiated terms.
```

File: packages/website/pages/docs/concepts/proposing.mdx [hunk 2]
```diff
@@ -8,86 +8,28 @@ On Taiko, the next L2 state is known immediately and deterministically at the ti
    - IF every transaction in the list is valid, an ordered subset of the list is created by skipping over transactions which have an invalid nonce or the sender has too little Ether balance to pay for the transaction. This ordered subset is used along with the [anchor transaction](./proposing#anchor-transaction) to create a Taiko L2 block.
    - IF any transaction in the list is invalid, an empty block (with only the anchor tx) is created on L2.
 
-## Intrinsic validity functions
+## Building the blocks
 
-The Ethereum yellow paper has a well defined set of rules to compute the state transition. We use these same rules to take a proposed block, and compute the post-block state on Taiko. A proposed block on Taiko has two parts:
+The Ethereum yellow paper has a well defined set of rules to compute the state transition. We use these same rules to take a proposed block, and compute the post-block state on Taiko. The high level overview of creating L2 blocks as follows:
```
Comment (dantaik) [inline packages/website/pages/docs/concepts/proposing.mdx:13]:
```suggestion
he Ethereum yellow paper has well-defined rules to compute the state transition.. We use these same rules to take a proposed block, and compute the post-block state on Taiko. The high level overview of creating L2 blocks as follows:
```

File: packages/website/pages/docs/concepts/proposing.mdx [hunk 3]
```diff
@@ -8,86 +8,28 @@ On Taiko, the next L2 state is known immediately and deterministically at the ti
...
-2. The transaction's signature is valid (rule #2 in Ethereum yellow paper).
-3. The transaction's gas limit is no smaller than the intrinsic gas $K_{TxMinGasLimit}$ (rule #5 in the Ethereum yellow paper).
-
-If any of these fails, an empty block will be created on L2 (with only the anchor tx).
+1. The system starts by creating a new L2 block with an anchor transaction. This anchor transaction is always the first transaction in the block even if the block is empty. (More about anchor transactions in the next section.)
+2. There are validity checks which are performed on node (or sometimes on protocol) level as well:
+2.1 Asserts that the length of the transaction list does not exceed a predefined maximum (MAX_TX_LIST_BYTES), ensuring that the list is within the limits.
+2.2 The transaction list is (RLP) decoded into a list of transactions (txList). If the bytes are not decodeable it will result in an empty block.
+2.3 The amount of gas required to include transactions are available. (If a transaction cannot fit into a block, it is simply excluded from it, but the block is still valid with other transactions.)
```
Comment (dantaik) [inline packages/website/pages/docs/concepts/proposing.mdx:19]:
```suggestion
2.3 The amount of gas required to include transactions is available. (If a transaction cannot fit into a block, it is simply excluded from it, but the block is still valid with other transactions.)
```

File: packages/website/pages/docs/concepts/proposing.mdx [hunk 4]
```diff
@@ -8,86 +8,28 @@ On Taiko, the next L2 state is known immediately and deterministically at the ti
...
+2.4 The transaction signature is valid.
+2.5 The transaction nonce is valid.
+2.6 The sender account has no contract code deployed (EIP-3607).
+2.7 The transaction's gas limit is no smaller than the intrinsic gas.
+2.8. The sender has enought balance to cover the transaction (gasLimit * gasPrice + tx.value).
```
Comment (dantaik) [inline packages/website/pages/docs/concepts/proposing.mdx:24]:
```suggestion
2.8. The sender has enough balance to cover the transaction (gasLimit * gasPrice + tx.value).
```

**Review by dantaik — DISMISSED** [review]

File: packages/website/pages/docs/concepts/proposing.mdx [hunk 5]
```diff
@@ -8,86 +8,28 @@ On Taiko, the next L2 state is known immediately and deterministically at the ti
...
-   $$
-   \begin{aligned}
-   B &≡ (H, T, U) \\
-   &≡ M(δ, θ, B, \dot{B}) \\
-   &≡ M(δ, θ, C, L)
-   \end{aligned}
-   $$
+1. Persisting `l1Height`, `l1Hash`, `l1SignalRoot`, and `parentGasUsed` (data inherited from L1) to the storage trie. These values can be used by bridges to validate cross-chain messages.
+2. Comparing the public input hash stored by the previous block, with the hash of the previous 255 block hashes. The anchor transaction will throw an exception if such comparison fails. The protocol requires the anchor transaction to execute successfully and will not accept a proof for a block that fails to do so. Note that the genesis block has $ρ_0$ ≡ `KEC(0, d, [0, ..., 0])`.
+3. Persisting a new public input hash to the storage trie for the next block to use. This allows transactions, in the current and all following blocks, to access these public input data with confidence as their values are now covered by ZK-EVM’s storage proof.
```
Comment (Brechtpd) [inline packages/website/pages/docs/concepts/proposing.mdx:34]: I think this one can be removed as it's part of 2. I believe.

**Review by dantaik — APPROVED** [review]

**Review by Brechtpd — APPROVED** [review]

**Pull request merged**